src/dir_contents.ml

```diff
@@ -567,9 +567,14 @@ let rec get sctx ~dir =
       in
       Hashtbl.add cache dir t;
       t
-    | Is_component_of_a_group_but_not_the_root _ ->
-      (* Filled while scanning the group root *)
-      Option.value_exn (Hashtbl.find cache dir)
+    | Is_component_of_a_group_but_not_the_root _ -> begin
+        match Hashtbl.find cache dir with
+        | Some t -> t
+        | None ->
+          ignore (get sctx ~dir:(Path.parent_exn dir) : t);
```

Comment:
it seems a bit strange to call `get` for its side effects only. What do you think about adding a function that does the same as `get` but does nothing? (effectively giving a name to ` fun ... -> ignore (get ...)`)

Reply:
There is a comment on the next line, so I guess it's not too bad. I think this code could be refactored, it feels a bit messy, but that's more work

Reply:
I think that when we ignore the result of a get, the intent that we just need a side effect is already pretty clear. Sometimes the side effect can sometimes be unclear, but here in this case it's pretty clear I think.